accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -616,6 +609,442 @@ It is a warning to use a null aware operator (`?.`, `?[]`, `?..`, `??`, `??=`, o
 It is a warning to use the null check operator (`!`) on an expression of type
 `T` if `T` is **strictly non-nullable** .
 
+### Local variables and definite (un)assignment.
+
+As part of the null safety release, errors for local variables are specified to
+take into account **definite assignment** and **definite unassignment** (see the
+section on Definite Assignment below).  We say that a variable is **potentially
+assigned** if it is not **definitely unassigned**, and that a variable is
+**potentially unassigned** if it is not **definitely assigned**.
```

Comment:
I was very confused about the two "potential" states for a while because I couldn't tell how they differed. That's because I assumed that these states were mutually exclusive. But on further reader, I guess that's not the intent? Am I correct that these mean:

* Definitely assigned: All paths assign the variable.
* Definitely assigned: No path assigns the variable.
* Potentially assigned: There exists at least one path which assigns the variable. There may be paths that do not, or it may be that all paths do. So every definitely assigned variable is also potentially assigned.
* Potentially unassigned: There exists at least one path which does not assign the variable. There may be paths that do, or it may be that no paths do. So every definitely unassigned variable is also potentially unassigned.

If the "potentially" states are intended to be exclusive of the "definitely" states, then I think that implies that there is only one potentially state:

* Either: There exists at least one path which assigns (to exclude it from the definitely unassigned state) and at least one path which does not assign (to exclude it from the definitely assigned state).

Am I understanding this correctly?

Reply:
Yes, these are not exclusive.  I introduced these terms in response to feedback that just using "not definitely assigned" (i.e. could be "Either" or could be "Definitely unassigned") was confusing.

Reply:
I think those terms are helpful for the overall readability.